### PR TITLE
add tests for exception classes

### PR DESCRIPTION
**Pull request by mtrberzi**
addresses issue #52

Something is very fishy with the code coverage report, as it says that for all the exception classes I'm testing, I'm not calling the getMessage() method, even though the tests very clearly do this...

### CODE REVIEW

**Comment by mtrberzi** [comment]
I keep forgetting that without Flowdock you'll have to check manually to see when I have a pull request. Here's one.

**Comment by mtrberzi** [comment]
Give me a minute to see why the build failed -- in the meantime @lucaswoj can you check #54 and merge? I'd like to get that into master for consistency with this outstanding branch

**Comment by mtrberzi** [comment]
Okay, sorry about that, this should now use Expressions correctly. Forgot the FacadeExpression class was removed.

**Comment by lucaswoj** [comment]
Does jacoco report full coverage of these files now?

**Comment by mtrberzi** [comment]
No, my copy is still showing missed coverage on getMessage()

**Comment by lucaswoj** [comment]
Ok. I'll take a look at this after I get off work!

**Comment by lucaswoj** [comment]
I'm seeing the test coverage. I had to invoke `jacoco` as 

```
gradle build jacocoTestReport
```

though. Perhaps there's something janky about our configuration right now.

File: src/test/java/org/whdl/frontend/syntaxtree/TestMultipleAssignmentException.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    return new LiteralExpression(BitTypeValue.getInstance());
+  }
+  
+  private Variable v_inst = null; 
+  public Variable getVariableInstance(){
+    if(v_inst == null){
+      v_inst = new Variable(getVariableIdentifierInstance(), getTypeExpression());
+    }
+    return v_inst;
+  }
```
Comment (lucaswoj) [inline src/test/java/org/whdl/frontend/syntaxtree/TestMultipleAssignmentException.java:27]: Why do you need to lazily create the variable instance? I think our convention so far has been that the `get*Instance()` methods always create a new instance and tests that require the same instance multiple times just hold a local copy.

Comment (lucaswoj) [inline src/test/java/org/whdl/frontend/syntaxtree/TestMultipleAssignmentException.java:27]: Also, test names are the _only_ place where we use underscores in identifiers like this. This property should be in `camelCase`

File: src/test/java/org/whdl/frontend/syntaxtree/TestVariableNotDefinedException.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  private NamespaceIdentifier getNamespaceIdentifierInstance() {
+    if (namespaceIdentifierInstance == null) {
+      ArrayList<String> name = new ArrayList<String>(1);
+      name.add("whdl");
+
+      namespaceIdentifierInstance = new NamespaceIdentifier(name);
+    }
+
+    return namespaceIdentifierInstance;
+  }
```
Comment (lucaswoj) [inline src/test/java/org/whdl/frontend/syntaxtree/TestVariableNotDefinedException.java:15]: Same feedback as above :smile:

File: src/test/java/org/whdl/frontend/syntaxtree/TestVariableNotAssignedException.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    return new LiteralExpression(BitTypeValue.getInstance());
+  }
+  
+  private Variable v_inst = null; 
+  public Variable getVariableInstance(){
+    if(v_inst == null){
+      v_inst = new Variable(getVariableIdentifierInstance(), getTypeExpression());
+    }
+    return v_inst;
+  }
```
Comment (lucaswoj) [inline src/test/java/org/whdl/frontend/syntaxtree/TestVariableNotAssignedException.java:27]: Same feedback as above

File: src/test/java/org/whdl/frontend/syntaxtree/TestVariableNotAssignedException.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  private NamespaceIdentifier getNamespaceIdentifierInstance() {
+    if (namespaceIdentifierInstance == null) {
+      ArrayList<String> name = new ArrayList<String>(1);
+      name.add("whdl");
+
+      namespaceIdentifierInstance = new NamespaceIdentifier(name);
+    }
+
+    return namespaceIdentifierInstance;
+  }
```
Comment (lucaswoj) [inline src/test/java/org/whdl/frontend/syntaxtree/TestVariableNotAssignedException.java:15]: Same feedback as above

File: src/test/java/org/whdl/frontend/syntaxtree/TestMultipleDefinitionException.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  private NamespaceIdentifier getNamespaceIdentifierInstance() {
+    if (namespaceIdentifierInstance == null) {
+      ArrayList<String> name = new ArrayList<String>(1);
+      name.add("whdl");
+
+      namespaceIdentifierInstance = new NamespaceIdentifier(name);
+    }
+
+    return namespaceIdentifierInstance;
+  }
```
Comment (lucaswoj) [inline src/test/java/org/whdl/frontend/syntaxtree/TestMultipleDefinitionException.java:15]: Same feedback as above

**Comment by lucaswoj** [comment]
@MurphEngineer Does my feedback make sense? Glad to have a discussion about it, if you'd like. Would love to have this merged!

**Comment by mtrberzi** [comment]
Sure. I'll look into this one in a bit...

**Comment by mtrberzi** [comment]
Okay, this commit cleans up all of the things you mentioned, was there anything else?

**Comment by lucaswoj** [comment]
Nope! This looks great!

**Pull request merged**